src/Exception/InvalidPhoneNumberException.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\I18n\PhoneNumber\Exception;
+
+use InvalidArgumentException;
+
+use function sprintf;
+
+final class InvalidPhoneNumberException extends InvalidArgumentException implements ExceptionInterface
+{
+    /** @psalm-pure */
+    public static function with(string $phoneNumber, ?string $countryCode): self
```

Comment:
I'm wondering if this should be split into 3 separate types:

- InvalidPhoneNumberException
- InvalidPhoneNumberForRegionException
- UndetectablePhoneNumberRegionException

These could have a shared marker interface they all implement, if there's a desire to catch more generically.

I'm finding that when an exception can be used for multiple distinct error types, during testing, we have to validate that against expected messages, and that becomes problematic when the message changes. Having discrete exception types simplifies the testing, gives more granularity when catching, and, if a marker interface is used to group exceptions by "type", you can catch more generically if desired.

More classes, but better flexibility.

Reply:
Thanks @weierophinney - this also revealed some dead code for removal too 👍